Assets/Scripts/Game/EnemySenses.cs

```diff
@@ -184,8 +183,37 @@ public float TargetRateOfApproach
             set { targetRateOfApproach = value; }
         }
 
+        public float LastHadLOSTimer
+        {
+            get { return lastHadLOSTimer; }
+            set { lastHadLOSTimer = value; }
+        }
+
+
+
+        //Delegates to allow mods to replace or extend senses logic.
+        //Mods can potentially save the original value before replacing it, if access to default behaviour is still desired.
+        public delegate bool BlockedByIllusionEffectCallback();
+        public BlockedByIllusionEffectCallback BlockedByIllusionEffect { get; set; }
+
+        public delegate bool CanSeeTargetCallback(DaggerfallEntityBehaviour target);
+        public CanSeeTargetCallback CanSeeTarget { get; set; }
+
+        public delegate bool CanHearTargetCallback();
+        public CanHearTargetCallback CanHearTarget { get; set; }
+
+        public delegate bool CanSeePlayerLightCallback();
+        public CanSeePlayerLightCallback CanSeePlayerLight { get; set; }
```

Comment:
I like the extra feature for detecting players, but I don't see why it should have the name "light". It's probably how your mod is going to use it. However, other mods might run other kinds of checks there.

I'm fine with it if you can rename it to something like "CanSeePlayerExtra" or "CanSeePlayerOther"

Reply:
Yes, that is oddly specific.  I'm changing it to CanDetectPlayerOther().

Reply:
I changed it again to CanDetectOtherwise and removed the player-specific check.